### ISSUE TITLE
Easier way to disable type hints in output

### ISSUE THREAD

**Issue by borekb** [yargs/yargs#969]
There's a previous question about `[boolean]` in `--help` and `--version` #319, with this suggested workaround:

```
yargs.getOptions().boolean.splice(-2);
```

However, this feels like a workaround indeed :) How about some global switch like this?

```
yargs.disableTypesInOutput().etc;
```

**Comment by bcoe** [comment]
@borekb I'm reluctant to add any more toggles to yargs, as it starts to complicate the API and makes it hard to sift through help.

 Out of curiosity, what's your reason for not wanting to display the type hints? In my opinion this provides a really useful hint to your users.

**Comment by borekb** [comment]
Hi @bcoe, I have two motivations:

1. `--version` and `--help` are not really perceived as booleans by the users so I'd rather not confuse them or make them wonder whether they need to provide `--version=true`.
2. Most CLI apps don't provide type hints in their help output and I'd like to use yargs to build a CLI that looks "standard". Easy way to turn types off feels natural.

Hope this helps explain it a bit.

**Comment by bcoe** [comment]
I can point you at where to change this code, if you want to perhaps publish a scoped fork of yargs with no usage messages. I think this is edge-case enough, that I'd rather not land this toggle into the main codebase.

**Comment by borekb** [comment]
I don't think I'm going to be maintaining a fork just for this :) Is the workaround from OP reliable in your opinion? If so, I'm happy with it but I don't understand what that code is doing and whether it won't break if I add more boolean options. Maybe another way would be to avoid `.help()` and `.version()` and provide my own type-less alternatives? Thanks for any hints.

**Comment by borekb** [comment]
Maybe I could re-implement the built in help command, I never looked at how to do that.. At least I would maintain just this small "fork" (I expect copy & paste of some internal function of yargs).

**Comment by borekb** [comment]
That sounds good! I agree that yargs is [complex enough](http://yargs.js.org/docs/) already, I was starting with it two weeks back and first-time experience was relatively convoluted. So 👍 for keeping it simple and finding other ways to customize it.

**Comment by binarymist** [comment]
I have to agree with @borekb, it doesn't make sense to have a type for an option that doesn't require a type. Custom options can be detyped by simply adding an empty string, it seems that `version` and `help` can not be though https://github.com/yargs/yargs/issues/319#issuecomment-395266705

**Comment by BigstickCarpet** [comment]
Here's another workaround that works for me.  The `yargs.parse()` method takes an optional callback argument.  If you specify a callback function, then Yargs will **not** display any output.  Instead, it passes the output as a string to the callback function.  You can then modify the string to remove the types.

```javascript
  function yargsCallback(err, args, output) {
    if (output) {
      // Remove types (e.g. [string], [boolean]) from the output
      output = output.replace(/\[\w+\]/g, '');
      
      // Show the modified output
      console.log(output);
    }
  }
```
